client/cli/src/commands/runcmd.rs

```diff
@@ -329,18 +330,18 @@ impl RunCmd {
 		let keyring = self.get_keyring();
 		let is_dev = self.shared_params.dev;
 		let is_light = self.light;
-		let is_authority = (self.validator || self.sentry || is_dev || keyring.is_some())
+		let is_authority = (self.validator || self.sentry.is_some() || is_dev || keyring.is_some())
 			&& !is_light;
 		let role =
 			if is_light {
-				sc_service::Roles::LIGHT
+				sc_service::Role::Light
 			} else if is_authority {
-				sc_service::Roles::AUTHORITY
+				sc_service::Role::Authority { sentry_nodes: Vec::new(), }
 			} else {
-				sc_service::Roles::FULL
+				sc_service::Role::Full
```

Comment:
We can probably just set the role to `Sentry` here, with the knowledge that it will be announced to the network as `Authority`.